sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -671,6 +674,57 @@ public String getId() {
         return withContext(context -> replaceItemInternal(itemType, itemId, doc, requestOptions, context));
     }
 
+    /**
+     * Run patch operations on an Item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single Cosmos item response with the replaced item.
+     *
+     * @param <T> the type parameter.
+     * @param itemId the item id.
+     * @param partitionKey the partition key.
+     * @param patchOperations patchOperations.
+     * @param itemType the item type.
+     * @return an {@link Mono} containing the Cosmos item resource response with the replaced item or an error.
+     */
+    public <T> Mono<CosmosItemResponse<T>> patchItem(
+        String itemId,
+        PartitionKey partitionKey,
+        List<PatchOperation<?>> patchOperations,
+        Class<T> itemType) {
+        return patchItem(itemId, partitionKey, patchOperations, new CosmosItemRequestOptions(), itemType);
+    }
+
+    /**
+     * Run patch operations on an Item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a single Cosmos item response with the replaced item.
+     *
+     * @param <T> the type parameter.
+     * @param itemId the item id.
+     * @param partitionKey the partition key.
+     * @param patchOperations patchOperations.
+     * @param options the request options.
+     * @param itemType the item type.
+     * @return an {@link Mono} containing the Cosmos item resource response with the replaced item or an error.
+     */
+    public <T> Mono<CosmosItemResponse<T>> patchItem(
+        String itemId,
+        PartitionKey partitionKey,
+        List<PatchOperation<?>> patchOperations,
+        CosmosItemRequestOptions options,
+        Class<T> itemType) {
+
+        if (options == null) {
+            options = new CosmosItemRequestOptions();
+        }
+        ModelBridgeInternal.setPartitionKey(options, partitionKey);
```

Comment:
I assume this will validate that pk is not null, else do it.

Reply:
Now validating partition key as it is required.